content/blog/2025-03-24-datafusion-46.0.0.md

```diff
@@ -0,0 +1,92 @@
+- **Ultra-fast `uuid()` UDF:** The `uuid()` function (which generates random UUID strings) received a major speed-up. It’s now roughly **40× faster** than before! The new implementation avoids unnecessary string copying and uses a more direct conversion to hex, making bulk UUID generation far more practical (PR [#14675](https://github.com/apache/datafusion/pull/14675) by [@simonvandel](https://github.com/simonvandel)).
+- **Accelerated `chr()` and `to_hex()`:** Several scalar functions have been micro-optimized. The `chr()` function (which returns the character for a given ASCII code) is about **4× faster** now, and the `to_hex()` function (which converts numbers to hex string) is roughly **2× faster**. These improvements may be most noticeable in tight loops or when these functions are applied to large arrays of values (PR [#14700](https://github.com/apache/datafusion/pull/14700) for `chr`, [#14686](https://github.com/apache/datafusion/pull/14686) for `to_hex` by [@simonvandel](https://github.com/simonvandel)).
+- **No More RowConverter in Grouped Ordering:** We removed an inefficient step in the *partial grouping* algorithm. The `GroupOrderingPartial` operator no longer converts data to “row format” for each batch (via `RowConverter`). Instead, it uses a direct arrow-based approach to detect sort key changes. This eliminated overhead and yields a nice speedup for certain aggregation queries. (PR [#14566](https://github.com/apache/datafusion/pull/14566) by [@ctsk](https://github.com/ctsk)).
+- **Predicate Pruning for `NOT LIKE`:** DataFusion’s parquet reader can now prune row groups using `NOT LIKE` filters, similar to how it handles `LIKE`. This means if you have a filter such as `column NOT LIKE 'prefix%'`, DataFusion can use min/max statistics to skip reading files/parts that can be determined to either entirely match or not match the predicate. In particular, a pattern like `NOT LIKE 'X%'` can skip data ranges that definitely start with "X". While a niche case, it contributes to query efficiency in those scenarios (PR [#14567](https://github.com/apache/datafusion/pull/14567) by [@UBarney](https://github.com/UBarney)).
+- **`UNION [ALL|DISTINCT] BY NAME` Support:** While not a pure speed optimization, it’s worth noting we added support for `UNION BY NAME` (both `UNION BY NAME` which is distinct by default, and `UNION ALL BY NAME`). This allows combining two result sets by aligning columns by name instead of by position, which can save you from manual `SELECT` reordering and thus simplify query logic. The new logical plan builder functions `union_by_name()`and `union_by_name_distinct()` correspond to these operations. This feature, inspired by systems like Spark and DuckDB, makes it easier to union heterogenous datasets. (PR [#14538](https://github.com/apache/datafusion/pull/14538) by [@rkrishn7](https://github.com/rkrishn7))
```

Comment:
I think these last two items are more features that performance -- maybe we can list them under a new heading of "New Features" or something